### DISCUSSION TITLE
In UI enable display of target metrics using Prometheus server as proxy

### DISCUSSION THREAD

**Discussion by bluepuma77** [prometheus/prometheus#10288] [Ideas]
## Proposal
In Prometheus UI enable display of target metrics using Prometheus server as proxy

## Problem
The Prometheus UI presents a nice list of targets. The first column is always the metrics URL to be able to inspect them. 

The issue is that the server network is usually very restricted or everything is working within a k8s or Swarm network and your browser is not part of it. So you normally are not able to access the metrics URL from the browser directly. 

## A possible solution
Provide an option (icon next to the URL?) that fetches the metrics using the Prometheus server as proxy. 

It does not necessarily need a real UI, it could just open a new window and display metrics in plain text via a URL like http://prometheus:9090/api/v1/target_metrics?url=http://10.10.1.139:8082/metrics.

<img width="800" alt="Prometheus UI" src="https://user-images.githubusercontent.com/5221187/148388494-b1d958ee-fbf7-40c8-9b0e-306c1408bb99.png">

**Comment by roidelapluie** [comment]
This would turn Prometheus into a reverse proxy, which does not seem desirable. I think we could add an button in the "labels" that would go to the graph page with the labels pre-selected but it would not really represent a scrape as extra metrics could be added by recording rules or metrics could be changed by metric_relabel_configs.

Therefore I do not think this issue is really actionnable for us.

**Comment by nirfr** [comment]
Hi, I have an "in between" idea:
The target is registered as a tag. So the icon can use a link which holds a simple url, without a need for a reverse proxy, so target 10.10.1.139:8082 is translated to
http://prometheus:9090/graph?g0.expr={instance%3D%2210.10.1.139:8082"}

So the change is a short javascript code
